dali/pipeline/operators/detection/box_encoder.cc

```diff
@@ -93,26 +94,59 @@ vector<BoundingBox> BoxEncoder<CPUBackend>::ReadBoxesFromInput(
   return boxes;
 }
 
-void BoxEncoder<CPUBackend>::WriteBoxToOutput(const BoundingBox &box, float *out_box_data) const {
-  const auto out_box = box.AsCenterWh();
-
+void BoxEncoder<CPUBackend>::WriteBoxToOutput(const std::array<float, BoundingBox::kSize> &box, float *out_box_data) const {
   for (unsigned idx = 0; idx < BoundingBox::kSize; ++idx)
-    out_box_data[idx] = out_box[idx];
+    out_box_data[idx] = box[idx];
 }
 
 void BoxEncoder<CPUBackend>::WriteAnchorsToOutput(float *out_boxes, int *out_labels) const {
-  for (unsigned idx = 0; idx < anchors_.size(); ++idx) {
-    WriteBoxToOutput(anchors_[idx], out_boxes + idx * BoundingBox::kSize);
-    out_labels[idx] = 0;
+  if (offset_) {
+    std::memset(out_boxes, 0, sizeof(std::remove_pointer<decltype(out_boxes)>::type)*anchors_.size());
```

Comment:
I think third param is 4 times too small. `anchors_size()` is number of anchors and each anchor is 4 floats. Additionally, maybe it would be more readable to just do: `sizeof(float*BoundingBox::kSize*anchors_.size())` since we know this is a pointer to floats.

Reply:
Done